### ISSUE TITLE
[java] false positives on Lombok @Value class fields

### ISSUE THREAD

**Issue by john-karp** [pmd/pmd#2458]
**Affects PMD Version:** 
6.23.0

**Rule:**
<https://pmd.github.io/latest/pmd_rules_java_codestyle.html#defaultpackage>
<https://pmd.github.io/latest/pmd_rules_java_codestyle.html#commentdefaultaccessmodifier>

**Description:**

Lombok makes fields of classes annotated with `@Value` `private final` by default. See
https://projectlombok.org/features/Value

**Code Sample demonstrating the issue:**

```
package com.test;

import lombok.Value;

@Value
public class ValueClass {
    String foo;
}
```

**Expected outcome:**

There shouldn't be any violations reported; the rule should not apply to classes annotated with `@Value`.

**Running PMD through:** *CLI*

**Comment by jsotuyod** [comment]
`CommentDefaultAccessModifier` has supported `@lombok.Value` since PMD 6.52.0 (see #4201)

PMD 7.0.0 completely removed `DefaultPackage`, as it was already fully covered by `CommentDefaultAccessModifier`
